src/middlewared/middlewared/plugins/pool_/pool.py

```diff
@@ -231,10 +228,8 @@ async def __convert_topology_to_vdevs(self, topology):
                     vdev['draid_data_disks'] = t_vdev['draid_data_disks']
                     vdev['draid_spare_disks'] = t_vdev['draid_spare_disks']
                 vdevs.append(vdev)
-                # cache and log devices should not have a swap
-                create_swap = True if i == 'data' else False
                 for disk in t_vdev['disks']:
-                    disks[disk] = {'vdev': vdev_devs_list, 'create_swap': create_swap}
+                    disks[disk] = {'vdev': vdev_devs_list, 'create_swap': False}
```

Comment:
No need to keep `create_swap` around ?

Reply:
Same answer as I gave in https://github.com/truenas/middleware/pull/13695#discussion_r1600408780